sycl/doc/extensions/proposed/sycl_ext_intel_register_mode.asciidoc

```diff
@@ -0,0 +1,142 @@
+  sycl_ext_oneapi_properties]
+* link:../experimental/sycl_ext_oneapi_kernel_properties.asciidoc[
+  sycl_ext_oneapi_kernel_properties]
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+
+== Backend support status
+
+This extension is currently implemented in {dpcpp} only for Intel GPU devices and
+only when using the Level Zero backend or OpenCL backend.
+Attempting to use this extension in kernels that run on other devices or
+backends may result in undefined behavior.  Be aware that the compiler
+is not able to issue a diagnostic to warn you if this happens.
+
+== Overview
+
+There are devices where the size of the general register file (GRF) used by a kernel is 
+configurable. Developers may want to change the GRF size based on their
+application. This extension adds the kernel property `register_mode` which provides a way
+to specify the GRF size.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_REGISTER_MODE` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Register Mode Property
+
+The `register_mode` kernel property provides a way to specify the GRF size used by a kernel. 
+It accepts a single integer value. It is defined as follows:
+```c++
+namespace sycl::ext::intel::experimental {
+
+struct register_mode_key {
+  template <int Mode>
+  using value_t = 
+      oneapi::experimental::property_value<register_mode_key, 
+                                           std::integral_constant<int, Mode>>;
+};
+
+template <int Mode>
+inline constexpr register_mode_key::value_t<Mode> register_mode;
+
+} // namespace sycl::ext::intel::experimental
+```
+The supported values are as follows:
+[%header,cols="1,5"]
+|===
+|GPU |Supported Values
+| PVC | 128, 256
+| DG2 | 128, 256
+|===
+
+Providing a value not consistent with the supported values may result in undefined behavior.
+
+=== Using the property in a kernel
+
+A simple example of using this extension is below.
+
+```c++
+using namespace sycl::ext::intel::experimental;
+{
+  ...
+  properties kernel_properties{register_mode<256>};
```

Comment:
```suggestion
namespace syclex = sycl::ext::oneapi::experimental;
namespace intelex = sycl::ext::intel::experimental;
{
  ...
  syclex::properties kernel_properties{intelex::register_mode<256>};
```

Our BKM for these examples is to define a namespace alias like `syclex`, so it's clear which APIs are in which namespaces.  This is particularly useful in this example since the `properties` and `register_mode` identifiers are in different namespaces.

Reply:
added in latest commit, thanks